src/main/java/com/faforever/client/map/MapService.java

```diff
@@ -424,6 +438,11 @@ public void evictCache() {
   }
 
   private CompletableFuture<Void> downloadAndInstallMap(String folderName, URL downloadUrl, @Nullable DoubleProperty progressProperty, @Nullable StringProperty titleProperty) {
+    if (mapGeneratorService.isGeneratedMap(folderName)) {
+      return mapGeneratorService.generateMap(folderName).thenRun(() -> {
```

Comment:
Please remove the empty theRun, no idea what it is for but there are certainly better ways for this

Reply:
If it is needed to match <Void> just make generateMap be <Void>

Reply:
Maybe it is even ok like that but considerng that the then run probably starts a new Runnable on the Completed Future thread pool that seems to be quite annoying

Reply:
do you have a different solution, it's basically about converting the string return type to void, I don't see an alternative